Quickstart example fails

**Describe the bug**
The quickstart example fails with the following error:

```
npm run mcp

...
    return new TSError(diagnosticText, diagnosticCodes, diagnostics);
           ^
TSError: ⨯ Unable to compile TypeScript:
src/mcp/index.ts:22:17 - error TS2339: Property 'name' does not exist on type 'RequestHandlerExtra'.

22   async (uri, { name }) => ({
                   ~~~~

    at createTSError (/Users/antontroynikov/Projects/CascadeProjects/panopticon/node_modules/ts-node/src/index.ts:859:12)
    at reportTSError (/Users/antontroynikov/Projects/CascadeProjects/panopticon/node_modules/ts-node/src/index.ts:863:19)
    at getOutput (/Users/antontroynikov/Projects/CascadeProjects/panopticon/node_modules/ts-node/src/index.ts:1077:36)
    at Object.compile (/Users/antontroynikov/Projects/CascadeProjects/panopticon/node_modules/ts-node/src/index.ts:1433:41)
    at Module.m._compile (/Users/antontroynikov/Projects/CascadeProjects/panopticon/node_modules/ts-node/src/index.ts:1617:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/antontroynikov/Projects/CascadeProjects/panopticon/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:1288:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1104:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:173:12) {
  diagnosticCodes: [ 2339 ]
}
```

`RequestHandlerExtra` does indeed have no such property:

https://github.com/modelcontextprotocol/typescript-sdk/blob/438505b74ceaacfb7b28ab9e877c728ba5e19479/src/shared/protocol.ts#L72

**To Reproduce**

Copy the v 1.3.0 quickstart example into an index.ts file, and then do `npm run mcp` 

**Expected behavior**
The example should succesfully run. 

**Logs**
If applicable, add logs to help explain your problem.

**Additional context**
Installed via `npm install @modelcontextprotocol/sdk`

--- a/README.md
+++ b/README.md
@@ -42,7 +42,7 @@ npm install @modelcontextprotocol/sdk
 Let's create a simple MCP server that exposes a calculator tool and some data:
 
 ```typescript
-import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
+import { McpServer, ResourceTemplate } from "@modelcontextprotocol/sdk/server/mcp.js";
 import { z } from "zod";
 
 // Create an MCP server
@@ -62,7 +62,7 @@ server.tool("add",
 // Add a dynamic greeting resource
 server.resource(
   "greeting",
-  "greeting://{name}",
+  new ResourceTemplate("greeting://{name}", { list: undefined }),
   async (uri, { name }) => ({
     contents: [{
       uri: uri.href,
@@ -114,7 +114,7 @@ server.resource(
 // Dynamic resource with parameters
 server.resource(
   "user-profile",
-  "users://{userId}/profile",
+  new ResourceTemplate("users://{userId}/profile", { list: undefined }),
   async (uri, { userId }) => ({
     contents: [{
       uri: uri.href,
@@ -185,7 +185,7 @@ server.prompt(
 A simple server demonstrating resources, tools, and prompts:
 
 ```typescript
-import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
+import { McpServer, ResourceTemplate } from "@modelcontextprotocol/sdk/server/mcp.js";
 import { z } from "zod";
 
 const server = new McpServer({
@@ -195,7 +195,7 @@ const server = new McpServer({
 
 server.resource(
   "echo",
-  "echo://{message}",
+  new ResourceTemplate("echo://{message}", { list: undefined }),
   async (uri, { message }) => ({
     contents: [{
       uri: uri.href,

